### PR TITLE
unblock proto update check

### PR DESCRIPTION
**Pull request by egaxhaj**
1. Checks `go.mod` for version changes in `cosmos-sdk` and `tendermint` libraries
2. Pulls down the Protobuf files for those versions and runs a `diff` against the Protobuf files in `main`
3. Logs error message with recommended steps to update to the latest Protobuf files and exits workflow.
4. If no update is necessary
    - runs`buf lint`
    - runs `buf breaking`

closes: #842 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: scripts/proto-update-check.sh [hunk 1]
```diff
@@ -6,23 +6,39 @@ set -ex
...
+    echo -e "Found differences in Protobuf files.\n
+    This indicates a version change was detected in one of the following libraries:
+      - github.com/cosmos/cosmos-sdk
+      - github.com/tendermint/tendermint
+
+    Review the diff log above and update accordingly. Perform the following steps locally.
+      1. run: make proto-update-deps
+      2. run: make proto-update-check
+      3. run: diff -rq -x '*.yaml' --exclude=google build/third_party third_party
+      4. commit updates and push\n" >&2
```
Comment (dwedul-figure) [inline scripts/proto-update-check.sh:35]: Here also with the `echo`, `printf`, and `cat << EOF` thing.
```suggestion
    printf '\n\nDiff log:\n\n%s\n\n" "$DIFF"

    cat << EOF >&2
Found differences in Protobuf files.
This indicates a version change was detected in one of the following libraries:
  - github.com/cosmos/cosmos-sdk
  - github.com/tendermint/tendermint

Review the diff log above and update accordingly. Perform the following steps locally.
  1. run: make proto-update-deps
  2. make sure this then passes: make proto-update-check
  3. commit updates and push
EOF
```

**Review by dwedul-figure — COMMENTED** [review]
There's a few things that would be nice to have updated for better portability and clarity.

File: .github/workflows/proto.yml [hunk 2]
```diff
@@ -20,6 +27,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - name: Setup go
         uses: actions/setup-go@v3
         with:
```
[the remark this answers is not in the document]
Comment (dwedul-figure) [reply .github/workflows/proto.yml:34]: I just discoverd `if: always()`, and kind of like that over firing up separate runners for each simple check.
This is what I ended up with: [PR 845's proto.yml](https://github.com/provenance-io/provenance/blob/7c91b77be06529288a563eb4c4e2450be2533998/.github/workflows/proto.yml).

Why did you add the `fetch-depth: 0` here?

File: scripts/proto-update-check.sh [hunk 3]
```diff
@@ -6,23 +6,39 @@ set -ex
...
+# single brackets because of `set -e` option above.
+if [ "$output" -gt "0" ]; then
+  echo -e "Downloading latest third_party proto files for comparison..."
+
+  # Download third_party proto files int build/ directory for comparison against $dir /third_party
+  bash "$dir"/scripts/proto-update-deps.sh build
+
+  echo -e "\nChecking Protobuf files for differences...\n"
+
+  DIFF=$(diff -rq -x '*.yaml' --exclude=google "$dir"/build/third_party "$dir"/third_party || true)
```
Comment (dwedul-figure) [inline scripts/proto-update-check.sh:22]: So yeah.... This is a significantly better solution than the one I had in mind. My thoughts were overly complicated and would probably be prone to breaking.

👍 🥇 💯

**Comment by dwedul-figure** [comment]
The `third_party breaking` step that is part of the `Protobuf Checks` workflow is currently failing because the `GrantAuthorization` message definition was moved from `proto/cosmos/authz/v1beta1/genesis.proto` to `proto/cosmos/authz/v1beta1/authz.proto`. However, neither file is new, and both are part of the same package, so it's not actually breaking anything.

I double checked this by regenerating the java/kotlin code (`cd protoBindings; ./gradlew assemble`), which resulted in the same generated code as before.

So in this PR, that specific failure can be ignored.

**Review by dwedul-figure — APPROVED** [review]

**Review by derekadams — APPROVED** [review]

**Pull request merged**